addons/product/models/product_product.py

```diff
@@ -175,7 +175,7 @@ def _check_barcode_unique(self):
         # Collect `product.product.id` of objects that share a barcode with any of those in `self`,
         # and group them by barcode.
         candidate_duplicates = self.env['product.product'].sudo().read_group(
-            domain=[("barcode", "in", list(set(product.barcode for product in self if product.barcode)))],
+            domain=[("barcode", "in", list(set(product.barcode for product in self if product.barcode))), ("active", "=?", False)],
```

Comment:
I don't understand the use of =? the ? should work if the value is False or NULL, but the = will work with False, so how are active=True records gotten?

if the intention is to only get both active and non-active product, you could try to add `.with_context(active_test=False)`  before the read group.

Reply:
Yes it's exactly what I want to do, thanks!